lib/validation.js

```diff
@@ -122,10 +122,24 @@ function wrapValidationError (result, dataVar) {
   return error
 }
 
+// no idea about performance
```

Comment:
The "idea" would be _slow_.

Reply:
This will be slow unfortunately. We need a different solution for this.